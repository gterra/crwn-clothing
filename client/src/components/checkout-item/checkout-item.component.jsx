import React from "react";
import { connect } from "react-redux";

import {
  clearItemFromCart,
  addCartItem,
  removeCartItem,
} from "../../redux/cart/cart.actions";

import {
  TextSpan,
  CheckoutItemDiv,
  ImageDiv,
  QuantitySpan,
  RemoveButton,
} from "./checkout-item.styles";

const CheckoutItem = ({
  item,
  clearItemFromCart,
  addCartItem,
  removeCartItem,
}) => (
  <CheckoutItemDiv>
    <ImageDiv>
      <img src={item.imageUrl} alt="item" />
    </ImageDiv>
    <TextSpan>{item.name}</TextSpan>
    <QuantitySpan>
      <div onClick={() => removeCartItem(item)}>&#10094;</div>
      <span>{item.quantity}</span>
      <div onClick={() => addCartItem(item)}>&#10095;</div>
    </QuantitySpan>
    <TextSpan>{item.price}</TextSpan>
    <RemoveButton onClick={() => clearItemFromCart(item)}>
      &#10005;
    </RemoveButton>
  </CheckoutItemDiv>
);

const mapDispatchToProps = (dispach) => ({
  clearItemFromCart: (item) => dispach(clearItemFromCart(item)),
  addCartItem: (item) => dispach(addCartItem(item)),
  removeCartItem: (item) => dispach(removeCartItem(item)),
});

export default connect(null, mapDispatchToProps)(CheckoutItem);
