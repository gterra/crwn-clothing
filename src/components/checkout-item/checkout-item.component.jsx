import React from "react";
import { connect } from "react-redux";

import "./checkout-item.styles.scss";
import {
  clearItemFromCart,
  addCartItem,
  removeCartItem,
} from "../../redux/cart/cart.actions";

const CheckoutItem = ({
  item,
  clearItemFromCart,
  addCartItem,
  removeCartItem,
}) => (
  <div className="checkout-item">
    <div className="image-container">
      <img src={item.imageUrl} alt="item" />
    </div>
    <span className="name">{item.name}</span>
    <span className="quantity">
      <div className="arrow" onClick={() => removeCartItem(item)}>
        &#10094;
      </div>
      <span className="value">{item.quantity}</span>
      <div className="arrow" onClick={() => addCartItem(item)}>
        &#10095;
      </div>
    </span>
    <span className="price">{item.price}</span>
    <span className="remove-button" onClick={() => clearItemFromCart(item)}>
      &#10005;
    </span>
  </div>
);

const mapDispatchToProps = (dispach) => ({
  clearItemFromCart: (item) => dispach(clearItemFromCart(item)),
  addCartItem: (item) => dispach(addCartItem(item)),
  removeCartItem: (item) => dispach(removeCartItem(item)),
});

export default connect(null, mapDispatchToProps)(CheckoutItem);
