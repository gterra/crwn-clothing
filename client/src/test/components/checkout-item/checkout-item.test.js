import React from "react";
import { shallow } from "enzyme";

import { CheckoutItem } from "../../../components/checkout-item/checkout-item.component";

describe("CheckoutItem component", () => {
  let wrapper;
  let mockClearItem;
  let mockAddItem;
  let mockRemoveItem;

  beforeEach(() => {
    mockClearItem = jest.fn();
    mockAddItem = jest.fn();
    mockRemoveItem = jest.fn();

    const mockProps = {
      item: {
        imageUrl: "www.testImage.com",
        price: 10,
        name: "hats",
        quantity: 2,
      },
      clearItemFromCart: mockClearItem,
      addCartItem: mockAddItem,
      removeCartItem: mockRemoveItem,
    };

    wrapper = shallow(<CheckoutItem {...mockProps} />);
  });

  it("should render CheckoutItem component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call clearItem when remove button is clicked", () => {
    wrapper.find("RemoveButton").simulate("click");
    expect(mockClearItem).toHaveBeenCalled();
  });

  it("should call removeItem when left arrow is clicked", () => {
    wrapper.find("QuantitySpan").childAt(0).simulate("click");

    expect(mockRemoveItem).toHaveBeenCalled();
  });

  it("should call addItem when right arrow is clicked", () => {
    wrapper.find("QuantitySpan").childAt(2).simulate("click");

    expect(mockAddItem).toHaveBeenCalled();
  });
});
