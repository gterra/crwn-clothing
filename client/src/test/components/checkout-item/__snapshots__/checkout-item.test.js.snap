// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutItem component should render CheckoutItem component 1`] = `
<styled.div>
  <styled.div>
    <img
      alt="item"
      src="www.testImage.com"
    />
  </styled.div>
  <styled.span>
    hats
  </styled.span>
  <QuantitySpan>
    <div
      onClick={[Function]}
    >
      ❮
    </div>
    <span>
      2
    </span>
    <div
      onClick={[Function]}
    >
      ❯
    </div>
  </QuantitySpan>
  <styled.span>
    10
  </styled.span>
  <RemoveButton
    onClick={[Function]}
  >
    ✕
  </RemoveButton>
</styled.div>
`;
